cmd/proxy/proxy.go

```diff
@@ -28,135 +28,108 @@ import (
 	"github.com/containous/traefik/v2/pkg/server/service"
 	traefiktls "github.com/containous/traefik/v2/pkg/tls"
 	"github.com/containous/traefik/v2/pkg/types"
-	"github.com/coreos/go-systemd/daemon"
 	"github.com/sirupsen/logrus"
 	"github.com/vulcand/oxy/roundrobin"
 )
 
 // NewCmd builds a new Proxy command.
 func NewCmd(loaders []cli.ResourceLoader) *cli.Command {
-	tConfig := cmd.NewTraefikConfiguration()
+	proxyConfig := cmd.NewProxyConfiguration()
 
 	return &cli.Command{
 		Name:          "proxy",
 		Description:   `Proxy command.`,
-		Configuration: tConfig,
+		Configuration: proxyConfig,
 		Run: func(_ []string) error {
-			return runCmd(&tConfig.Configuration)
+			return runCmd(proxyConfig)
 		},
 		Resources: loaders,
 	}
 }
 
-func runCmd(staticConfiguration *static.Configuration) error {
-	configureLogging(staticConfiguration)
+func runCmd(proxyConfiguration *cmd.ProxyConfiguration) error {
+	proxyConfiguration.Configuration.SetEffectiveConfiguration()
+
+	if err := proxyConfiguration.Configuration.ValidateConfiguration(); err != nil {
+		return err
+	}
+
+	configureLogging(&proxyConfiguration.Configuration)
 
 	http.DefaultTransport.(*http.Transport).Proxy = http.ProxyFromEnvironment
 
 	if err := roundrobin.SetDefaultWeight(0); err != nil {
 		log.WithoutContext().Errorf("Could not set round robin default weight: %v", err)
 	}
 
-	staticConfiguration.SetEffectiveConfiguration()
-
-	if err := staticConfiguration.ValidateConfiguration(); err != nil {
-		return err
-	}
-
 	log.WithoutContext().Infof("Maesh proxy version %s built on %s", version.Version, version.Date)
 
-	jsonConf, err := json.Marshal(staticConfiguration)
+	jsonConf, err := json.Marshal(proxyConfiguration.Configuration)
 	if err != nil {
 		log.WithoutContext().Errorf("Could not marshal static configuration: %v", err)
-		log.WithoutContext().Debugf("Static configuration loaded [struct] %#v", staticConfiguration)
-	} else {
-		log.WithoutContext().Debugf("Static configuration loaded %s", string(jsonConf))
+		log.WithoutContext().Debugf("Static configuration loaded [struct] %#v", proxyConfiguration.Configuration)
+
+		return err
 	}
 
-	svr, err := setupServer(staticConfiguration)
+	log.WithoutContext().Debugf("Static configuration loaded %s", string(jsonConf))
+
+	svr, err := setupServer(proxyConfiguration)
 	if err != nil {
 		return err
 	}
 
 	ctx := cmd.ContextWithSignal(context.Background())
 
-	if staticConfiguration.Ping != nil {
```

Comment:
The CLI flag in the Helm chart here: https://github.com/containous/maesh/blob/master/helm/chart/maesh/templates/mesh/mesh-daemonset.yaml#L157 should be removed too.

Reply:
This was removed by accident, It has been replaced and updated.

Reply:
I thought it was unused but it is by the readiness service.